lib/facter/kmod.rb

```diff
@@ -0,0 +1,29 @@
+Facter.add(:kmods) do
+  confine :kernel => 'Linux'
+  confine File.exists?('/sys/module')
+
+  kmod = {}
+
+  setcode do
+    Dir.foreach('/sys/module') do |directory|
+      next if directory == '.' or directory == '..'
+
+      kmod[directory] = {}
+      kmod[directory]['parameters'] = {}
+      kmod[directory]['used_by'] = []
```

Comment:
This could could be initialized in one go:

```ruby
kmod[directory] = {
  :parameters => {},
  :used_by    => [],
}
```

Reply:
Using symbols is also more elegant than strings for `:parameters` and `:used_by`